onnxruntime/core/providers/cuda/tensor/split_impl.cu

```diff
@@ -157,5 +157,67 @@ Status SplitImpl(cudaStream_t stream, const size_t element_size, const int block
   return Status::OK();
 }
 
+template <typename T>
+__global__ void _Split3InnerKernel(const int64_t size0,
```

Comment:
pls use vectorized loads

Reply:
Hi, @wangyems. when i trying to use vectorized loads, i found it will make the code more complex, because it must assure that the size is algin with vector size, and the code should handle other sizes.

Then i found a function call cuda::memcpy_async which can do async copy from memory to memory. i think it works here. Most of the cuda docs show this function as using to copy from global memory to shared memory, but i try to use it to copy from global memory to global memory, it works and performance is better than this hand written code.

I don't know if you are familiar with this cuda::memcpy_async function, is it ok for using there copy from global memory to global memory?

Reply:
@tianleiwu @wangyems I have implemented the vectorized version, please take a look this code.